openshift-fluentd.conf

```diff
@@ -0,0 +1,80 @@
+<system>
+  log_level warn
+</system>
+
+@include syslog-input-filter.conf
+
+<source>
+  @type tail
+  path "#{ENV['DATA_DIR']}/docker/*.log"
+  pos_file "#{ENV['DATA_DIR']}/es-containers.log.pos"
+  time_format %Y-%m-%dT%H:%M:%S
+  tag kubernetes.*
+  format json
+  keep_time_key true
+  read_from_head true
+</source>
+
+# simulate this filter
+# <filter kubernetes.**>
+#   type kubernetes_metadata
+#   kubernetes_url "#{ENV['K8S_HOST_URL']}"
+#   bearer_token_file /var/run/secrets/kubernetes.io/serviceaccount/token
+#   ca_file /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+#   include_namespace_id true
+# </filter>
+<filter kubernetes.**>
+  type record_transformer
+  enable_ruby
+  <record>
+    kubernetes_namespace_name ${tag_parts[3].split('_')[1]}
+    kubernetes_namespace_id 1
+    docker_container_id ${tag_parts[3].split('_')[2].split('-')[-1]}
+    kubernetes_pod_name ${tag_parts[3].split('_')[0]}
+    kubernetes_container_name ${tag_parts[3].split('_')[2].rpartition('-')[0]}
+    kubernetes_host localhost
+  </record>
+</filter>
+
+<filter kubernetes.**>
+  type record_transformer
+  enable_ruby
+  <record>
+    hostname ${(kubernetes_host rescue nil) || File.open('/etc/docker-hostname') { |f| f.readline }.rstrip}
+    message ${log}
+    version 1.1.4
+  </record>
+  remove_keys log,stream
+</filter>
+
+<match kubernetes.**>
+      @type elasticsearch_dynamic
+      host "#{ENV['ES_HOST']}"
+      port "#{ENV['ES_PORT']}"
+      scheme http
+      index_name ${record['kubernetes_namespace_name']}.${record['kubernetes_namespace_id']}.${Time.at(time).getutc.strftime(@logstash_dateformat)}
+
+      client_key "#{ENV['ES_CLIENT_KEY']}"
+      client_cert "#{ENV['ES_CLIENT_CERT']}"
+      ca_file "#{ENV['ES_CA']}"
+
+      flush_interval 5s
+      max_retry_wait 300
+      disable_retry_limit
+</match>
+
+<match system.** **_default_** **_openshift_** **_openshift-infra_**>
```

Comment:
I think it's important to have a match for fluentd.** as well, not to miss important errors/warnings from fluentd.